sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -138,4 +138,11 @@ object StaticSQLConf {
     .intConf
     .checkValue(_ >= 0, "Must be set greater or equal to zero")
     .createWithDefault(Int.MaxValue)
+
+  val OPTIMIZER_NON_EXCLUDABLE_RULES = buildStaticConf("spark.sql.optimizer.nonExcludedRules")
```

Comment:
I think `spark.sql.optimizer.staticRules` or `spark.sql.optimizer.unremovableRules` looks easy to understand.